bats/remotes-file-system.bats

```diff
@@ -117,3 +117,77 @@ SQL
     [[ "$output" =~ "remotes/anything/master" ]] || false
     [[ "$output" =~ "remotes/something/master" ]] || false
 }
+
+@test "fetch displays and updates branch list" {
+    # create a new branch
+    run dolt checkout -b tester
+    [ "$status" -eq 0 ]
+
+    # demonstrate there is no table named test on tester
+    run dolt ls
+    [ "$status" -eq 0 ]
+    [[ "${lines[0]}" =~ "No tables in working set" ]] || false
+
+    # seed tester with some data
+    dolt sql <<SQL
+CREATE TABLE test (
+  pk BIGINT NOT NULL COMMENT 'tag:0',
+  c1 BIGINT COMMENT 'tag:1',
+  c2 BIGINT COMMENT 'tag:2',
+  c3 BIGINT COMMENT 'tag:3',
+  c4 BIGINT COMMENT 'tag:4',
+  c5 BIGINT COMMENT 'tag:5',
+  PRIMARY KEY (pk)
+);
+SQL
+    dolt add test
+    dolt commit -m "test commit"
+
+    # create the remote data storage directories
+    mkdir remote1
+
+    # push both branches to remote
+    dolt remote add origin file://remote1
+    dolt push origin tester
+
+    dolt checkout master
+    dolt push origin master
+
+    # clone from a directory
+    cd dolt-repo-clones
+    dolt clone file://../remote1 test-repo
+    cd test-repo
+
+    run dolt branch
+    [ "$status" -eq 0 ]
+    [[ "${lines[0]}" =~ "master" ]] || false
+    [[ "${lines[1]}" =~ "tester" ]] || false
+
+    # delete tester branch
+    dolt branch -d -f tester
+
+    run dolt branch
+    [ "$status" -eq 0 ]
+    [[ "${lines[0]}" =~ "master" ]] || false
+    [[ "${lines[1]}" != "tester" ]] || false
+
+    # fetch branches from remote
+    dolt fetch
+
+    # should display both branches fetched from remote
+    run dolt branch
+    [ "$status" -eq 0 ]
+    [[ "${lines[0]}" =~ "master" ]] || false
+    skip "Fetched branch not displayed in branch list" [[ "${lines[1]}" =~ "tester" ]] || false
+
+    # should not be able to create branch with same
+    # name as remote branch
+    skip "Allows creation of branch with same name as remote branch" run dolt checkout -b tester
+    skip "Allows creation of branch with same name as remote branch" [ "$status" -eq 1 ]
+    skip "Allows creation of branch with same name as remote branch" [[ "${lines[0]}" =~ "fatal: A branch named 'tester' already exists." ]] || false
+
+    # should not be able to checkout branch that is not displayed
+    skip "Allows checkout of branch not displayed in branch list" run dolt checkout tester
```

Comment:
once the bug is fixed, the checks on 191 & 192 will fail because `tester` will be fetched on 175